src/index.js

```diff
@@ -176,37 +182,49 @@ function addMetadata({ types: t }) {
       CallExpression(babelPath, state) {
         if (!state.opts.shouldLoadFile) return;
 
-        // If we're at a top-level call expression, then we reset the beforeEachModified state to false, and let Babel
-        // move on to the next call expression.
-        if (babelPath.scope.block.type === 'Program') {
-          state.opts.beforeEachModified = false;
-          return;
+        // If this call expression is a top-level module, store string arg, else skip the nested module entirely
+        if (babelPath.get('callee').isIdentifier({ name: 'module' })) {
+          if (babelPath.parentPath.parent.type === 'Program') {
+            state.opts.moduleName = babelPath.get('arguments')[0].node.value;
+          } else {
+            // skip traversing contents in this nested module
+            babelPath.skip();
+          }
         }
 
-        if (!state.opts.beforeEachModified) {
-          const hasBeforeEach = babelPath.node.callee.property
+        if (
+          state.opts.moduleName &&
+          !state.opts.transformedModules.includes(state.opts.moduleName)
+        ) {
+          const isBeforeEach = babelPath.node.callee.property
             ? babelPath.node.callee.property.name === 'beforeEach'
             : false;
-          const testMethodCalls = ['test', 'module'];
-
-          let nodeName = '';
-
-          if (babelPath.node.callee.name) {
-            nodeName = babelPath.node.callee.name;
-          } else if (babelPath.node.callee.object) {
-            nodeName = babelPath.node.callee.object.name;
+          let isFirstChildTestMethodCall;
+
+          if (!isBeforeEach) {
+            const testMethodCalls = ['test', 'tests', 'module'];
```

Comment:
I'd probably move this out into a module scoped const, unless of course you're mutating this somewhere else (which I can't seem to see).